### PR TITLE
add DECFLOAT16 and DECFLOAT34

### LINKED ISSUE #788: decfloat16 and decfloat34 not supported by SET_CELL
decfloat16 and decfloat34 are predefined numeric types added in ABAP 7.02. They are currently not supported by method SET_CELL, an exception is raised.

Code to reproduce the error:
```
DATA: gc_save_file_name TYPE string VALUE 'test.xlsx'.
INCLUDE zdemo_excel_outputopt_incl.
START-OF-SELECTION.
  TRY.
      DATA(lo_excel) = NEW zcl_excel( ).
      DATA(lo_worksheet) = lo_excel->get_active_worksheet( ).
      lo_worksheet->set_cell( ip_column = 'A' ip_row = 1 ip_value = conv decfloat16( 1 ) ).
      lcl_output=>output( cl_excel = lo_excel ).
    CATCH cx_root INTO DATA(lx_root).
      MESSAGE lx_root TYPE 'I' DISPLAY LIKE 'E'.
  ENDTRY.
```

Is it possible to support them?

### PR DESCRIPTION
**Pull request by bigld**
closes #788

### CODE REVIEW

File: src/zcl_excel_converter.clas.abap [hunk 1]
```diff
@@ -1729,7 +1729,10 @@ CLASS zcl_excel_converter IMPLEMENTATION.
         r_format = wo_worksheet->get_default_excel_date_format( ).
       WHEN cl_abap_typedescr=>typekind_time.
         r_format = wo_worksheet->get_default_excel_time_format( ).
-      WHEN cl_abap_typedescr=>typekind_float OR cl_abap_typedescr=>typekind_packed.
+      WHEN cl_abap_typedescr=>typekind_float OR cl_abap_typedescr=>typekind_packed OR
+           cl_abap_typedescr=>typekind_decfloat OR
```
Comment (sandraros) [inline src/zcl_excel_converter.clas.abap:1733]: Not sure if the generic decfloat type is needed, but that doesn't harm.

File: src/zcl_excel_worksheet.clas.abap [hunk 2]
```diff
@@ -2946,7 +2946,10 @@ CLASS zcl_excel_worksheet IMPLEMENTATION.
               lv_value = zcl_excel_common=>number_to_excel_string( ip_value = <fs_numeric> ).
             ENDIF.
 
-          WHEN cl_abap_typedescr=>typekind_float OR cl_abap_typedescr=>typekind_packed.
+          WHEN cl_abap_typedescr=>typekind_float OR cl_abap_typedescr=>typekind_packed or
+               cl_abap_typedescr=>typekind_decfloat OR
```
Comment (sandraros) [inline src/zcl_excel_worksheet.clas.abap:2951]: same remark for generic type decfloat

**Review by sandraros — APPROVED** [review]
I did some tests with SET_CELL, converter, bind_alv and bind_table, all work fine.
LGTM 👍

**Comment by sandraros** [comment]
Could you merge the latest changes from master into your branch please? Thanks.

File: src/zcl_excel_converter.clas.abap (hunk 1 above)
Comment (bigld) [reply src/zcl_excel_converter.clas.abap:1733]: I'm also not sure but better to be careful now ;)

**Pull request merged**